Add initial WebView2 power user FAQ (#2521)
MicrosoftDocs/edge-developer: microsoft-edge/webview2/concepts/power-user-faq.md

Reviewer: it is traditional to not add new .md to toc.yml in 1st commit.  I will tag in VS Code & paste here...

Reviewer: added locally

---

> title: Power User FAQ

Reviewer: ```suggestion
title: WebView2 Power User FAQ
```

Reviewer: updated locally

> description: Answers to some frequently asked questions about WebView2 from non-developers who want to learn more about it.

Reviewer: ```suggestion
description: Tips for end users about how to use WebView2 apps.
```

Reviewer: updated locally

author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: conceptual
ms.prod: microsoft-edge
ms.technology: webview
ms.date: 03/29/2023
---

> # WebView2 Power User FAQ

Reviewer: ```suggestion
# Using WebView2 apps
```
I'd avoid framing as a "FAQ" and use standard topical framing (title, doc design) instead.  Design the title along with Sentence 1 that clarifies the meaning of the title.
alts:
Using WebView2 apps
End-users info about WebView2
Information about WebView2 for end users
WebView2 Power User FAQ
WebView2 End User FAQ

whichever title is selected, propagate to the usual locations:
* title field at top
* h1 title
* TOC.yml
* link text of any links in other pages that link to here

Reviewer: likely to keep as faq, per offline

> Most of our documentation is intended for developers who want to use WebView2 in their applications. This FAQ is intended for a normal person looking to understand what this `msedgewebview2.exe` application is on your computer. If we don't have your question here, feel free to hit "Feedback" in the top-right corner, or open a specific issue [on our GitHub](https://github.com/MicrosoftDocs/edge-developer/issues) and we'll see if we can add it.

Author: I'm not 100% sure about "normal person". "layperson" was suggested, but sounds similarly off?

Reviewer: fixed locally
WebView2 end-user FAQ

> Most of our documentation is intended for developers who want to use WebView2 in their applications. This FAQ is intended for a normal person looking to understand what this `msedgewebview2.exe` application is on your computer. If we don't have your question here, feel free to hit "Feedback" in the top-right corner, or open a specific issue [on our GitHub](https://github.com/MicrosoftDocs/edge-developer/issues) and we'll see if we can add it.

Reviewer: ```suggestion
Most of our documentation is intended for developers who want to use WebView2 in their applications. This FAQ is for end users, to explain the `msedgewebview2.exe` application on your computer. If we don't have your question here, feel free to hit "Feedback" in the top-right corner, or open a specific issue [on our GitHub](https://github.com/MicrosoftDocs/edge-developer/issues) and we'll see if we can add it.
```

Reviewer: fixed locally, as:
```
This FAQ is for end users, to explain the `msedgewebview2.exe` application on your computer.  If your question isn't listed here, you can click the **Feedback** button in the top-right corner, to enter a question as an [Issue in the edge-developer Docs repo](https://github.com/MicrosoftDocs/edge-developer/issues), and your question might be added to this article.  See also [Contact the WebView2 Team](../contact.md).
```

> ## What is WebView2?
> WebView2 is a way for any Windows app developer to embed web content in their application. This can let them write code for a website or web app, then re-use it in their Windows application, saving time and effort.

Reviewer: ```suggestion

<!-- ====================================================================== -->
## What is WebView2?

WebView2 is a way for any Windows app developer to embed web content in their application. Including the WebView2 control in an app lets developers write code for a website or web app, and then re-use that web code in their Windows application, saving time and effort.  See [Introduction to Microsoft Edge WebView2](../index.md).

```

> ## Why are there so many WebView2 processes running?
> WebView2 follows Microsoft Edge and [Chromium's process model](https://chromium.googlesource.com/chromium/src/+/main/docs/process_model_and_site_isolation.md). Having functionality broken up into multiple processes help reliability, security, and performance - Each process performs a certain responsibility, and can do that uninterrupted, even if one of the others hits a problem. There are normally a few processes - WebView2 manager, a GPU process, some utility processes (one for network, audio, etc. depending on the content), and some "renderer" processes. You'll have one set of processes per app using WebView2, and roughly one renderer process per WebView2 in the app, similar to how you have one renderer process per tab in a browser. See our [process model documentation]( https://learn.microsoft.com/microsoft-edge/webview2/concepts/process-model) for more details.

Reviewer: ```suggestion

<!-- ====================================================================== -->
## Why are there so many WebView2 processes running?

WebView2 follows the process model that's used in the Chromium browser engine of Microsoft Edge, described in [Process Model and Site Isolation](https://chromium.googlesource.com/chromium/src/+/main/docs/process_model_and_site_isolation.md).  Having functionality broken up into multiple processes help reliability, security, and performance.  Each process performs a certain responsibility, and can do that work uninterrupted, even if one of the other processes hits a problem.

There are normally a few processes:
*  The WebView2 manager.
*  A GPU process.
*  Utility processes, such as network or audio, depending on the content.
*  Renderer processes.

Your machine will have one set of processes for each app that uses WebView2, and typically one renderer process for each WebView2 control that's in the app, similar to how you have one renderer process per tab in a browser.  For details, see [Process model for WebView2 apps](/microsoft-edge/webview2/concepts/process-model).
```
rewrite of h2 section.

When you test that Learn link in rendered page, you might have to double-click "review" in Address bar, press Delete, press Enter.  if it initially shows 404.

> ## Why is WebView2 using so much memory, CPU, etc.?
> In most cases this is related to how WebView2 is being used, not WebView2 itself. If the application is rendering poorly optimized web content, then it may consume more resources. Task Manager will normally show the processes parented, but can be incorrect at times or if you sort by a column other than "Name". You can also use Microsoft’s [Process Explorer](https://learn.microsoft.com/sysinternals/downloads/process-explorer) tool. If there's an app consistently causing heavy resource usage you can follow up with the support for that app.

Reviewer: ```suggestion
<!-- ====================================================================== -->
## Why is WebView2 using so much memory or CPU resources?

In most cases, extra memory or CPU usage is due to how WebView2 is being used, not WebView2 itself. If the application is rendering non-optimized web content, then the WebView2 control might consume more resources. Task Manager will normally show the processes parented, but can be incorrect at times or if you sort by a column other than **Name**. 

You can also use Microsoft's [Process Explorer](/sysinternals/downloads/process-explorer) tool. If you find that an app is consistently causing heavy resource usage, you can contact Support for that app to report the resource usage.

```

> ## Is WebView2 a virus?
> WebView2 is not a virus. It is a Microsoft product that is updated automatically to ensure that your applications have the latest security fixes. Most people will see it used by Widgets, Teams, Office, Outlook, Weather, and many more applications. Any application can *use* WebView2 though, so it's possible for a malicious application to use WebView2, just like it can use any other Windows API or feature.

Reviewer: ```suggestion
<!-- ====================================================================== -->
## Is WebView2 a virus?

WebView2 is not a virus.  WebView2 is a Microsoft product that is updated automatically (by using the _Evergreen_ distribution method) to ensure that your applications have the latest feature support and security fixes. 

Most people will see WebView2 used by Widgets, Teams, Office, Outlook, Weather, and many more applications. Any application can _use_ WebView2 though, so it's possible for a malicious application to use WebView2, just like it can use any other Windows API or feature.

```

> ## Will uninstalling Edge break WebView2?
> No, Edge can be uninstalled without causing a problem for WebView2's evergreen distribution method. When both Edge and WebView2 are installed, Edge's installer will link the applications together so that one doesn't use extra disk space.

Reviewer: ```suggestion
<!-- ====================================================================== -->
## Will uninstalling Microsoft Edge make WebView2 stop working?

No. Microsoft Edge can be uninstalled without causing a problem for WebView2.  When both Microsoft Edge and WebView2 are installed, the Microsoft Edge installer links the applications together, to avoid using extra disk space.

```

> ## I uninstalled WebView2, why is it back again?
> WebView2 is pre-installed on Windows 11 and optimistically installed on most Windows 10 devices ([details here](https://blogs.windows.com/msedgedev/2022/06/27/delivering-the-microsoft-edge-webview2-runtime-to-windows-10-consumers/)). Many apps built on WebView2 also perform checks to re-install WebView2 when being installed or launched. Enterprises may also push WebView2 to devices they manage.
> 

Reviewer: ```suggestion
<!-- ====================================================================== -->
## I uninstalled WebView2, yet it was re-installed

WebView2 is pre-installed on Windows 11 and is installed on most Windows 10 devices.  See [Delivering the Microsoft Edge WebView2 Runtime to Windows 10 Consumers](https://blogs.windows.com/msedgedev/2022/06/27/delivering-the-microsoft-edge-webview2-runtime-to-windows-10-consumers/).

Many apps that are built on WebView2 also perform checks to re-install WebView2 when the app is installed or launched. Enterprises can also push WebView2 to devices that they manage.

<!-- ====================================================================== -->
## See also

* [Introduction to Microsoft Edge WebView2](../index.md)
* [Contact the WebView2 Team](../contact.md)
```
